### PR TITLE
[Refactoring] JenkinsService refactoring

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

The `JenkinsService` class contains code which can be extracted in dedicated utility classes in order to reduce complexity.

### Description

In this PR we extract code from the `JenkinsService` class. We move all build plan related related code into the `BuildPlanService` class. We also create a `JenkinsBuildPlanUtils.java` which contains utility methods used by the build plan service.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a Programming Exercise
4. Participate in the programming exercise and make sure that you get the correct results.

### CODE REVIEW

**Review by willergeorg — DISMISSED** [review]
Tested on TS2. Everything worked.

**Review by NicolasUllmann — DISMISSED** [review]
Tested on TS1. Everything worked as expected.

**Review by sinangunduz — APPROVED** [review]
tested on ts1, works as expected

**Review by katjanakosic — DISMISSED** [review]
Works as expected. Tested on TS1.

**Review by canyelten — DISMISSED** [review]
Tested on TS2, works as expected.

File: src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanUtils.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    /**
+     * Replace old XML files that are not based on pipelines.
+     * Will be removed in the future
+     *
+     * @param jobXmlDocument The Jenkins job config xml
+     */
+    @Deprecated
+    public static void replaceRemoteURLs(Document jobXmlDocument, String repoUrl, String repoNameInCI) throws IllegalArgumentException {
```
Comment (FrankeLukas) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanUtils.java:59]: `repoUrl` and `repoNameInCI` don't have a `@param` so the server-style fails:
grafik
(Tbh I don't know why this wasn't a problem before copying the code around?)

Comment (ivanchimeno) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanUtils.java:59]: Thanks for the heads-up, I'll change that

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Still works. Re-approve.

**Review by real-dmy — APPROVED** [review]
Tested on TS2. Works as expected.

**Review by NicolasUllmann — APPROVED** [review]
Tested on TS2. Everything worked as expected.

**Pull request merged**